### DISCUSSION TITLE
How can I serialize com.ibm.watson.assistant.v1.model.Context?

### DISCUSSION THREAD

**Discussion by ruimo** [watson-developer-cloud/java-sdk#1159] [Q&A] [answered]
As Watson Assistant's context need to be propagated during a set of conversation, I need to serialize the WA context into external storage such as Redis or Browser cookie. However, com.ibm.watson.assistant.v1.model.Context does not implement java.io.Serializable, so I have no idea to serialize context. In the source code of Context, there are SerializedName annotation and it seems Context is designed serializable.

https://github.com/watson-developer-cloud/java-sdk/blob/master/assistant/src/main/java/com/ibm/watson/assistant/v1/model/Context.java

Why Context does not implement java.io.Serializable? Any workaround to serialize WA context?

**Comment by kevinkowa** [comment, marked as answer]
Hi @ruimo,

You can try the following `GsonSingleton.getGson().toJson(contextModel);`. Let me know if it works for you!

**Reply by ruimo** [reply]
Hi @kevinkowa ,

It worked like a charm.
Thank you for your info!
